extensions/amp-ad/0.1/test/test-amp-ad-custom.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpAdCustom} from '../amp-ad-custom';
+import {toggleExperiment} from '../../../../src/experiments';
+import * as sinon from 'sinon';
+
+describe('Amp custom ad', () => {
+  let sandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  /**
+   * Get a custom amp-ad element
+   * @param {string} url The url of the ad server
+   * @param {string} slot The alphanumeric slot Id (optional)
+   * @returns {Element} The completed amp-ad element, which has been added to
+   *    the current document body.
+   */
+  function getCustomAd(url, slot) {
+    toggleExperiment(window, 'ad-type-custom', true);
+    const ampAdElement = document.createElement('amp-ad');
+    ampAdElement.setAttribute('type', 'custom');
+    ampAdElement.setAttribute('width', '500');
+    ampAdElement.setAttribute('height', '60');
+    ampAdElement.setAttribute('data-test', 1);
+    ampAdElement.setAttribute('data-url', url);
+    if (slot) {
+      ampAdElement.setAttribute('data-slot', slot);
+    }
+    document.body.appendChild(ampAdElement);
+    return ampAdElement;
+  }
+
+  it('should get the correct full URLs', done => {
+    // Create all the ads *before* calling getFullUrl_() - otherwise, the
+    // ads after the first getFullUrl_() call will not be in the cache.
+
+    // Single ad with no slot
```

Comment:
Since you've already grouped them well, can you split them to 3 different test cases? i.e., 3 "it" clauses.